Feat/tox

jbarno opened the pull request:
Just test runner changes. This will enable test runs in both py2.7 and py2.6 (because of CentOS)

# TODO:
- [x] update readme
- [x] ensure travis is happy

Commit:
feat(tox): not really working but lets see how travis feels

Commit:
fix(coverage): took out coverage from codacy

Commit:
fix(tox): got to install it

Commit:
feat(tox): works locally dropped py3 testenvs

Commit:
feat(travis_tox): simplier travis

Commit:
chore(py26): keep py26 around not a good idea

However we really need to release note that in order for this to work on py26 we have to DOWNGRADE CRYPTOGRAPY

the definition of a terrible idea

ProfOak commented:
The travis tests for python2.7 seem like they execute twice but I'm not sure why.

Commit:
fix(pycparser): lets go back one and see py26

jbarno commented:
Yeah something is funky looks like its actually running 4 times.

Commit:
fix(tox): following the example for tox-travis

Commit:
fix(syntax_err): in library requests-mock

caused by https://github.com/jamielennox/requests-mock/commit/58e3e9c977a152411a5f6cf2cae37f2a70d77758 in py26

Commit:
fix(requests_mock): dont think this will work unpin

Commit:
feat(threadleak): add threadleak to pytest 2.7 and 2.6 runs

Commit:
fix(drop_py26): out of scope of just getting tox working

Commit:
chore(docs): run tests through tox in README

anmaxvl commented:
addressed by https://github.com/NCI-GDC/gdc-client/pull/248
